File: app/src/main/java/com/example/applecustomer/fusedlocationprovider/LocationUpdateBroadcastReceiver.java
package com.example.applecustomer.fusedlocationprovider;

import android.app.PendingIntent;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.location.Location;
import android.util.Log;

import com.google.android.gms.location.LocationResult;

import java.util.List;

import static android.content.Context.MODE_PRIVATE;

public class LocationUpdateBroadcastReceiver extends BroadcastReceiver {
    public static final String ACTION_PROCESS_UPDATES = "123";

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d("reciver","broadcast recivee");
        if (intent != null) {
            final String action = intent.getAction();
            if (ACTION_PROCESS_UPDATES.equals(action)) {
              /*  LocationResult result = LocationResult.extractResult(intent);
                if (result != null) {
                    Location location = result.getLastLocation();
                    String longitude = String.valueOf(location.getLongitude());
                    String latitude = String.valueOf(location.getLatitude());

                    SharedPreferences.Editor editor = context.getSharedPreferences("location",MODE_PRIVATE).edit();
                    editor.putString("lat", latitude);
                    editor.putString("lon",longitude);
                    editor.apply();*/

                Intent i = new Intent(context, LocationService.class);
                i.setAction(LocationService.ACTION_PROCESS_UPDATES);
                context.startService(i);

                }
            }
        }
    }

File: app/src/main/java/com/example/applecustomer/fusedlocationprovider/LocationUpdateBroadcastReceiver.kt
package com.example.applecustomer.fusedlocationprovider

import android.app.NotificationManager
import android.app.PendingIntent
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.location.Location
import android.media.RingtoneManager
import android.net.Uri
import android.support.v4.app.NotificationCompat
import android.util.Log

import com.google.android.gms.location.LocationResult

import android.content.Context.MODE_PRIVATE
import com.example.applecustomer.fusedlocationprovider.LocationService.Companion.ACTION_PROCESS_UPDATES

class LocationUpdateBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent?) {
        Log.d("reciver", "broadcast recivee")
        if (intent != null) {
            val action = intent.action
            if (ACTION_PROCESS_UPDATES == action) {
                /*val result = LocationResult.extractResult(intent)
                if (result != null) {
                    val location = result.lastLocation
                    val longitude = location.longitude.toString()
                    val latitude = location.latitude.toString()

                    val editor = context.getSharedPreferences("location", MODE_PRIVATE).edit()
                    editor.putString("lat", latitude)
                    editor.putString("lon", longitude)
                    showNotification(context, longitude, latitude)
                    editor.apply()*/


                val result = LocationResult.extractResult(intent)
                if (result != null) {
                    val location = result.lastLocation
                    val longitude = location.longitude.toString()
                    val latitude = location.latitude.toString()

                  showNotification(context,longitude,latitude)

                    var i = Intent(context, LocationService::class.java)
                    i.action = LocationService.ACTION_PROCESS_UPDATES
                    i.putExtra("lat",latitude)
                    i.putExtra("lon",longitude)
                    context.startService(i)

                }

            }

        }

    }

    fun showNotification(context: Context, longitutde: String, latitude: String) {
        Log.d("notification", "In notification")

        val defaultSoundUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
        val notificationBuilder = NotificationCompat.Builder(context)
                .setContentTitle("Locations")
                .setContentText("longitude = " + longitutde + "latitude = " + latitude)
                .setAutoCancel(true)
                .setSound(defaultSoundUri)
                .setSmallIcon(R.drawable.cast_ic_notification_small_icon)

        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager

        notificationManager.notify(1, notificationBuilder.build())
    }

    companion object {
        val ACTION_PROCESS_UPDATES = "123"
    }
}


